src/osdp_phy.c

```diff
@@ -343,6 +345,18 @@ int osdp_phy_check_packet(struct osdp_pd *pd, uint8_t *buf, int len,
 			pd->ephemeral_data[0] = OSDP_PD_NAK_SEQ_NUM;
 			return OSDP_ERR_PKT_FMT;
 		}
+	} else {
+		if (comp == 0) {
+			/**
+			 * Check for receiving a busy reply from the PD which would
+			 * have a sequence number of 0, come in an unsecured packet
+			 * of minimum length, and have the reply ID REPLY_BUSY.
+			 */
+			if ((pkt_len == 6) && (pkt->data[0] == REPLY_BUSY)) {
+				pd->seq_number -= 1;
+				return OSDP_ERR_PKT_NONE;
```

Comment:
There is scope for some optimisation here: since all action for this packet is complete, we don't need subsequent calls `osdp_phy_decode_packet()` and `cp_decode_response()` in caller.

This can be done by returning a new error code `OSDP_ERR_PKT_BUSY` here and translating that to `OSDP_CP_ERR_RETRY_CMD` in the caller.

Reply:
That is exactly what I had in the first and most tested variant of this change!  But I got to thinking that not having the extra error check would save a couple or so bytes of executable code size and that benefit for the size constrained application would outweigh the downside of the extra execution cycles for the probably seldom encountered busy reply.
```
	if (err == OSDP_ERR_PKT_WAIT) {
		/* rx_buf_len < pkt->len; wait for more data */
		return OSDP_CP_ERR_NO_DATA;
	}
	if (err == OSDP_ERR_PKT_BUSY) {
		/* rx_seq_number == 0 && reply_id ==  REPLY_BUSY; resend command */
		return OSDP_CP_ERR_RETRY_CMD;
	}
	if (err == OSDP_ERR_PKT_NONE) {
		/* Valid OSDP packet in buffer */
		len = osdp_phy_decode_packet(pd, pd->rx_buf, len, &buf);
		if (len <= 0) {
			return OSDP_CP_ERR_GENERIC;
		}
		err = cp_decode_response(pd, buf, len);
	}
```

Reply:
Fair point. Only for this case let's cheat and set `OSDP_ERR_PKT_BUSY` to 2 (in the enum define it above `OSDP_ERR_PKT_NONE` with a comment so we know its intentionally +ve). Caller needs no modification as we anyway need to remove processed bytes and err would be passed above.

Reply:
Clever!  And the change to return 2 instead of 0 resulted in no increase in the executable image size for an ARM Cortex-M0+ using the GCC ARM compiler.